iwyu_path_util.cc

```diff
@@ -167,20 +178,33 @@ string MakeAbsolutePath(const string& path) {
 string MakeAbsolutePath(const string& base_path, const string& relative_path) {
   llvm::SmallString<128> absolute_path(base_path.c_str());
   llvm::sys::path::append(absolute_path, relative_path);
-
+  
   return absolute_path.str();
 }
 
 string GetParentPath(const string& path) {
   llvm::StringRef parent = llvm::sys::path::parent_path(path);
-  return parent.str();
+  string result = parent.str();
+  // Ensure trailing slash on non-empty paths
+  if(!result.empty() && result[result.size() - 1] != '/' && result[result.size() - 1] != '\\')
```

Comment:
C++11 is not a problem, we're already bound by the same language standards as LLVM/Clang, so we're at C++~12.5 (GCC 4.8 and VS2013, IIRC).

> I did not want to canonicalize the file path, as that would be beyond the scope of
> this function. Put in a backslash-path and get a slash-path although you 
> only called GetParent?

You're right, that makes no sense.

> I'd rather remove this and have an extra function NormalizeFolderPath that calls
> NormalizeFilePath and adds the slash

Sounds promising!